onnxruntime/core/framework/graph_partitioner.cc

```diff
@@ -120,20 +103,50 @@ static Status GetCapabilityForEP(Graph& graph, KernelRegistryManager& kernel_reg
       if (!capability || !capability->sub_graph) {
         continue;
       }
-      AssignNodes(graph, *capability->sub_graph, current_ep.Type());
+
+      AssignNodes(graph, *capability->sub_graph, ep_type);
     }
 
+    const NodeIndex first_new_node = graph.MaxNodeIndex();
+
     // Perform layout transformation on the specific EP assigned graph
     bool modified = false;
     ORT_RETURN_IF_ERROR(transform_layout(graph, modified, current_ep));
 
```

Comment:
Just wondering, what if you call graph.Resolve() here? Will it be successful?

Reply:
It should be. The layout transformer does all the work to maintain the edges correctly so that Graph::Resolve is not needed (required in a minimal build).